core/src/main/java/jenkins/slaves/DefaultJnlpSlaveReceiver.java

```diff
@@ -27,7 +28,7 @@
 @Extension
 public class DefaultJnlpSlaveReceiver extends JnlpAgentReceiver {
```

Comment:
Hmmm this is approach is slightly different than what I had in mind when you mentioned refactoring the remoting module. I was wondering if we could do-away with the server-side implementing JNLP version specific protocol handlers entirely? ie. The master would just make one call to the remoting library and the implementation there would try protocol 3, 2, 1.

This way we can add protocols to the remoting library without needing to update the master, and makes the remoting  library more re-usable and out-of-the-box ready.

Regardless this is fine since it works.